File: JAVA/24_DFS&BFS/BOJ_02_2606_바이러스.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Queue;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class Main {

    static int n, m, result;
    static boolean[] visited;
    static ArrayList<Integer>[] list;
    static StringBuilder sb;

    public static void bfs(int idx) {
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(idx);
        visited[idx] = true;
        while (!queue.isEmpty()) {
            int tmp = queue.poll();
            if (list[tmp] != null) {
                for (int i : list[tmp]) {
                    if (!visited[i]) {
                        visited[i] = true;
                        queue.offer(i);
                        result++;
                    }
                }
            }
        }
    }

    @SuppressWarnings("unchecked")
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        sb = new StringBuilder();
        n = Integer.parseInt(br.readLine());
        m = Integer.parseInt(br.readLine());
        visited = new boolean[n + 1];
        list = new ArrayList[n + 1];
        for (int i = 1; i <= n; i++) {
            list[i] = new ArrayList<Integer>();
        }
        for (int i = 0; i < m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            list[x].add(y);
            list[y].add(x);
        }
        result = 0;
        bfs(1);
        System.out.println(result);
    }
}

File: C/24_DFS&BFS/BOJ_02_2606_바이러스.c
#include <stdio.h>
int n, result = 0, com[101][101], visited[101], cnt[101];
void dfs(int idx) {
	visited[idx] = 1;
	for (int i = 0; i < cnt[idx]; i++) {
		if (!visited[com[idx][i]]) {
			result++;
			dfs(com[idx][i]);
		}
	}
}
main() {
	int m;
	scanf("%d %d", &n, &m);
	while (m--) {
		int x, y;
		scanf("%d %d", &x, &y);
		com[x][cnt[x]++] = y;
		com[y][cnt[y]++] = x;
	}
	dfs(1);
	printf("%d", result);
}